Mathlib/CategoryTheory/Category/Cat/Monoidal.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2024 Nicolas Rolland. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolas Rolland
+-/
+import Mathlib.CategoryTheory.ChosenFiniteProducts
+/-!
+`Cat` is a category on its own, views categories as its objects and functors as its morphism.
+On one hand, we can build a cartesian product category out of a pair of categories.
+On the other hand `Cat`, as a category, comes with its own definition for an object to be a product.
+
+We verify here that the product *in* `Cat` is given by the usual product of categories viewed
+as an object.
+
+## Implementation notes
+
+We rely on `CategoryTheory.ChosenFiniteProducts` to provide the full monoidal structure
+-/
+
+universe v u
+
+namespace CategoryTheory.Cat
+
+open Limits
+
+/-- The chosen terminal object in `Cat`. -/
+abbrev OneCat : Cat := Cat.of (Discrete PUnit)
+
+/-- The chosen terminal object in `Cat` is terminal. -/
+def isTerminalPUnit : IsTerminal OneCat where
+  lift s := Functor.star s.pt
+  fac _ (j : Discrete PEmpty.{1}) := PEmpty.elim j.as
+  uniq s m _ :=  Functor.punit_ext' m (Functor.star s.pt)
+
```

Comment:
```suggestion
/-- The chosen terminal object in `Cat`. -/
abbrev OneCat : Cat.{v, u} := Cat.of (ULift (ULiftHom (Discrete Unit)))

/-- The chosen terminal object in `Cat` is terminal. -/
def isTerminalPUnit : IsTerminal OneCat :=
  IsTerminal.ofUniqueHom (fun _ ↦ (Functor.const _).obj ⟨⟨⟨⟩⟩⟩) fun _ _ ↦ rfl

```
This is to make everything universe-polymorphic (i.e. make it work for `Cat.{v, u}`).

Reply:
Thank you, how to manage universe properly is still a bit of a mystery to me for now.
 
And it is apparently big pb for code reuse. 

(In Mathlib/CategoryTheory/Functor/KanExtension/Basic.lean
`The bicategory API cannot be used directly here because it would not allow the universe
polymorphism which is necessary for some applications.`)

I should read more about this